### PR TITLE
[UNDERTOW-2245]javax.servlet.forward.mapping request attribute is not…

### PR DESCRIPTION
**Pull request by moulalis**
Issue: https://issues.redhat.com/browse/UNDERTOW-2245

### CODE REVIEW

File: servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherIncludeTestCase.java [hunk 1]
```diff
@@ -258,9 +267,28 @@ public void testParametersAreMerged() throws IOException {
...
+            HttpGet get = new HttpGet(DefaultServer.getDefaultServerURL() + "/servletContext/dispatchServletInclude");
+            HttpResponse result = client.execute(get);
+            assertEquals(StatusCodes.OK, result.getStatusLine().getStatusCode());
+            final String response = HttpClientUtils.readResponse(result);
+            //UNDERTOW-2245 javax.servlet.forward.mapping request attribute is not available in servlets forwarded with RequestDispatcher
+
+            assertTrue(DeploymentUtils.isContainString(response,"jakarta.servlet.include.context_path"));
+            assertTrue(DeploymentUtils.isContainString(response,"jakarta.servlet.include.servlet_path"));
+            assertTrue(DeploymentUtils.isContainString(response,"jakarta.servlet.include.request_uri"));
+            assertTrue(DeploymentUtils.isContainString(response,"jakarta.servlet.include.mapping"));
```
Comment (fl4via) [inline servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherIncludeTestCase.java:290]: missing path info and query string

**Review by fl4via — CHANGES_REQUESTED** [review]
Overall it looks very good, but there are some small fixes and cosmetic changes that need to be applied here.

**Comment by moulalis** [comment]
Hi @fl4via 
I've updated the PR. Please review.
Thanks.

**Comment by soul2zimate** [comment]
Hi @moulalis, the new added tests failed in CI. 
```
Error:  Failures: 
Error:    DispatcherForwardTestCase.testDisptacherServletForward:398 COM01003: Internal error: Assertion failure: Expected boolean value to be false
Error:    DispatcherIncludeTestCase.testDisptacherServletInclude:291 COM01003: Internal error: Assertion failure: Expected boolean value to be false
```

**Comment by moulalis** [comment]
> Hi @moulalis, the new added tests failed in CI.
> 
> ```
> Error:  Failures: 
> Error:    DispatcherForwardTestCase.testDisptacherServletForward:398 COM01003: Internal error: Assertion failure: Expected boolean value to be false
> Error:    DispatcherIncludeTestCase.testDisptacherServletInclude:291 COM01003: Internal error: Assertion failure: Expected boolean value to be false
> ```

Looking into it.

File: servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherForwardTestCase.java [hunk 2]
```diff
@@ -362,9 +373,29 @@ public void testParametersAreMergedButNotDuplicated() throws IOException {
...
+            HttpGet get = new HttpGet(DefaultServer.getDefaultServerURL() + "/servletContext/dispatchServlet");
+            HttpResponse result = client.execute(get);
+            assertEquals(StatusCodes.OK, result.getStatusLine().getStatusCode());
+            final String response = HttpClientUtils.readResponse(result);
+            //UNDERTOW-2245 javax.servlet.forward.mapping request attribute is not available in servlets forwarded with RequestDispatcher
+
+            assertTrue(DeploymentUtils.isContainString(response,"jakarta.servlet.forward.context_path"));
+            assertTrue(DeploymentUtils.isContainString(response,"jakarta.servlet.forward.servlet_path"));
+            assertTrue(DeploymentUtils.isContainString(response,"jakarta.servlet.forward.request_uri"));
+            assertTrue(DeploymentUtils.isContainString(response,"jakarta.servlet.forward.mapping"));
```
[the remark this answers is not in the document]
Comment (moulalis) [reply servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherForwardTestCase.java]: Hi @fl4via As discussed this is causing test failure, Spec need to review for this to fix.

File: servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherIncludeTestCase.java (hunk 1 above)
Comment (moulalis) [reply servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherIncludeTestCase.java]: Hi @fl4via As discussed this is causing test failure, Spec need to review for this to fix.

**Comment by moulalis** [comment]
@fl4via I've updated the PR. Please review it.
Thanks.

**Review by fl4via — APPROVED** [review]

File: servlet/src/test/java/io/undertow/servlet/test/dispatcher/util/DispatcherUtil.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
```
Comment (fl4via) [inline servlet/src/test/java/io/undertow/servlet/test/dispatcher/util/DispatcherUtil.java:3]: Please, update the year to 20223.

File: servlet/src/test/java/io/undertow/servlet/test/dispatcher/NextServlet.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package io.undertow.servlet.test.dispatcher;
```
Comment (fl4via) [inline servlet/src/test/java/io/undertow/servlet/test/dispatcher/NextServlet.java:1]: Missing license header.

**Review by fl4via — CHANGES_REQUESTED** [review]

File: servlet/src/test/java/io/undertow/servlet/test/dispatcher/util/DispatcherUtil.java (hunk 3 above)
Comment (moulalis) [reply servlet/src/test/java/io/undertow/servlet/test/dispatcher/util/DispatcherUtil.java]: @fl4via I've updated the changes to 2023.

File: servlet/src/test/java/io/undertow/servlet/test/dispatcher/NextServlet.java (hunk 4 above)
Comment (moulalis) [reply servlet/src/test/java/io/undertow/servlet/test/dispatcher/NextServlet.java]: @fl4via I've updated the changes.
Thanks.

File: servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherIncludeServlet.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package io.undertow.servlet.test.dispatcher;
```
Comment (soul2zimate) [inline servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherIncludeServlet.java:1]: This also needs a license header.

Comment (moulalis) [reply servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherIncludeServlet.java:18]: Done. Thanks

Comment (soul2zimate) [reply servlet/src/test/java/io/undertow/servlet/test/dispatcher/DispatcherIncludeServlet.java:18]: s/2014/2023

**Review by fl4via — APPROVED** [review]

**Pull request merged**